src/async_operator/map.rs

```diff
@@ -0,0 +1,30 @@
+use crate::AsyncOperator;
+use futures::FutureExt;
+
+/// Convert Error.
+#[derive(Debug, Clone, Copy)]
+pub struct Map<F, P> {
+    pub(super) inner: P,
+    pub(super) f: F,
+}
+
+impl<I, F, P, O, E> AsyncOperator<I> for Map<F, P>
+where
+    P: AsyncOperator<I, Error = E>,
+    F: FnMut(Result<P::Output, P::Error>) -> Result<O, E>,
```

Comment:
It is much more common to use `F: FnMut(P::Output) -> O` (i.e., only convert the ok value), since we already have `MapErr`.

Also, don't forget to add `map` method to `AsyncOperatorExt`.